### PR TITLE
While stmt fix

### PR DESCRIPTION
**Pull request by ovdiiuv**
This PR adds support for assignments in while-loops. It also enables to combine multiple assignments in the while condition by adding supprot for some logoical operators.

Fixes: #913

### CODE REVIEW

File: lib/Differentiator/BaseForwardModeVisitor.cpp [hunk 1]
```diff
@@ -1633,21 +1633,45 @@ StmtDiff BaseForwardModeVisitor::VisitWhileStmt(const WhileStmt* WS) {
...
+
+    if ((condBO && (condBO->isLogicalOp() || condBO->isAssignmentOp())) ||
+        condUO) {
+      auto condDiff = Visit(cond);
+      if (condDiff.getExpr_dx() &&
+          (!isUnusedResult(condDiff.getExpr_dx()) || condUO))
+        cond = BuildOp(BO_Comma, BuildParens(condDiff.getExpr_dx()),
+                       BuildParens(condDiff.getExpr()));
+      else
+        cond = condDiff.getExpr();
```
Comment (vgvassilev) [inline lib/Differentiator/BaseForwardModeVisitor.cpp:1667]: Can we add a test case covering that branch? Can you also add a comment before the complex if-stmt condition which are the cases we enter with there?

File: lib/Differentiator/BaseForwardModeVisitor.cpp [hunk 2]
```diff
@@ -1662,29 +1686,6 @@ StmtDiff BaseForwardModeVisitor::VisitWhileStmt(const WhileStmt* WS) {
...
-  //   ...
-  // }
-  //
-  // gets differentiated to,
-  //
-  // while (double b = a) {
-  //   double _d_b = _d_a;
-  //   ...
-  //   ...
-  // }
```
Comment (vgvassilev) [inline lib/Differentiator/BaseForwardModeVisitor.cpp:1682]: This comment was rather good. Why did we drop it?

Comment (ovdiiuv) [reply lib/Differentiator/BaseForwardModeVisitor.cpp:1682]: Now what it does to the `while (double b = a){}` is it creates variables `b` and `_d_b` before the while statement and assigns them the right values inside of the while condition, so the comment is no longer relevant.

Comment (vgvassilev) [reply lib/Differentiator/BaseForwardModeVisitor.cpp:1682]: Why not updating the comment rather than deleting it?

**Pull request closed without merging**